### PR TITLE
Update feature files to include acceptance test for runner

### PR DESCRIPTION
**Pull request by CarolineClark**
Linked to this PR: https://github.com/julianghionoiu/tdl-client-java/pull/1

### CODE REVIEW

File: runner/ChallengeSession.feature [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  Background:
+    Given There is a challenge server running on "localhost" port 8222
+    And journeyId is "aJourneyId"
+    And the challenge server exposes the following endpoints
+      | verb       | endpointEquals               | status   | responseBody                         | acceptHeader  |
+      | GET        | /availableActions/aJourneyId | 200      | Available actions coming from server | text/coloured |
+      | GET        | /roundDescription/aJourneyId | 200      | RoundID\nRound Description           | text/coloured |
+      | GET        | /journeyProgress/aJourneyId  | 200      | Journey progress coming from server  | text/coloured |
+
+    And the challenge server exposes the following endpoints
```
Comment (julianghionoiu) [inline runner/ChallengeSession.feature]: Why do we need this as a separate And?

File: runner/ChallengeSession.feature [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      | GET        | /status           | 200      | OK             |
+      | POST       | /notify           | 200      | ACK            |
+
+  # Business critical scenarios
+
+  Scenario: The server interaction
+    Given the action input comes from a provider returning "anySuccessful"
+    And the challenges folder is empty
+    When user starts client
+    Then the server interaction should contains the following lines:
```
Comment (julianghionoiu) [inline runner/ChallengeSession.feature]: Typo: should contains

File: runner/ChallengeSession.feature [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      | verb       | endpointEquals                | status  | responseBody             | acceptHeader  |
+      | GET        | /availableActions/aJourneyId  | 200     | No actions available.    | text/coloured |
+    When user starts client
+    Then the client should not ask the user for input
+
+  Scenario: Should exit if recording not available
+    Given recording server is returning error
+    When user starts client
+    Then the client should not ask the user for input
+    And the user is informed that they should start the recording
```
Comment (julianghionoiu) [inline runner/ChallengeSession.feature]: I think we should explicitly assert on the output

File: runner/ChallengeSession.feature [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+  Scenario: challenge server is returning a client error
+    Given the challenge server returns 400, response body "Nothing here" for all requests
```
Comment (julianghionoiu) [inline runner/ChallengeSession.feature]: The make this even more explicit, the response body could be "Error message coming from server".

**Review by julianghionoiu — COMMENTED** [review]
I think we are almost there.

File: runner/ChallengeSession.feature (hunk 1 above)
Comment (CarolineClark) [inline runner/ChallengeSession.feature:12]: One matches the url based on a regular expression, the other matches exactly. They require different methods in wiremock.
We can probably do a check to see if we detect if the url contains a regular expression, but that felt a bit error prone.

Comment (CarolineClark) [reply runner/ChallengeSession.feature:12]: Alternatively if we can match without a regular expression, then we can also get rid of this.

File: runner/ChallengeSession.feature (hunk 3 above)
Comment (CarolineClark) [reply runner/ChallengeSession.feature]: ok

File: runner/ChallengeSession.feature (hunk 4 above)
Comment (CarolineClark) [reply runner/ChallengeSession.feature]: Sounds good.

File: runner/ChallengeSession.feature (hunk 1 above)
Comment (julianghionoiu) [reply runner/ChallengeSession.feature:12]: Ok, got it

**Review by julianghionoiu — APPROVED** [review]

**Comment by CarolineClark** [comment]
This needs to have the version tagged just before/just after being merged with v0.20

**Review by julianghionoiu — APPROVED** [review]

**Pull request merged**